src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -456,6 +460,85 @@ class spell_sha_earthen_rage_proc_aura : public AuraScript
     }
 };
 
+// 61882 - Earthquake
+class spell_sha_earthquake : public SpellScript
+{
+    PrepareSpellScript(spell_sha_earthquake);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_AFTERSHOCK, SPELL_SHAMAN_AFTERSHOCK_ENERGIZE });
+    }
+
+    void HandleAfterCast()
+    {
+        int32 energize = *GetSpell()->GetPowerTypeCostAmount(POWER_MAELSTROM);
+
+        if (AuraEffect* aftershock = GetCaster()->GetAuraEffect(SPELL_SHAMAN_AFTERSHOCK, EFFECT_0))
+            if (roll_chance_f(aftershock->GetAmount()))
+                GetCaster()->CastSpell(GetCaster(), SPELL_SHAMAN_AFTERSHOCK_ENERGIZE, CastSpellExtraArgs(energize)
+                    .AddSpellBP0(energize));
+    }
+
+    void Register() override
+    {
+        AfterCast += SpellCastFn(spell_sha_earthquake::HandleAfterCast);
+    }
+};
+
+// 61882 - Earthquake aura
+class spell_sha_earthquake_aura : public AuraScript
+{
+    PrepareAuraScript(spell_sha_earthquake_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_EARTHQUAKE, SPELL_SHAMAN_EARTHQUAKE_TICK });
+    }
+
+    void HandlePeriodic(AuraEffect const* /*aurEff*/)
+    {
+        if (Unit* caster = GetCaster())
+        {
+            std::vector<AreaTrigger*> areaTriggers = caster->GetAreaTriggers(SPELL_SHAMAN_EARTHQUAKE);
+
+            for (AreaTrigger* at : areaTriggers)
+                caster->CastSpell(at->GetPosition(), SPELL_SHAMAN_EARTHQUAKE_TICK, true);
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_sha_earthquake_aura::HandlePeriodic, EFFECT_1, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 77478 - Earthquake tick
+class spell_sha_earthquake_tick : public SpellScript
+{
+    PrepareSpellScript(spell_sha_earthquake_tick);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        if (!spellInfo->GetEffects().empty())
+            const_cast<SpellEffectInfo*>(&spellInfo->GetEffect(EFFECT_0))->BonusCoefficient = 0.391f;
+
+        return ValidateSpellInfo({ SPELL_SHAMAN_EARTHQUAKE_KNOCKING_DOWN });
+    }
+
+    void HandleOnHit()
+    {
+        if (Unit* target = GetHitUnit())
+            if (roll_chance_i(GetEffectInfo(EFFECT_1).BasePoints))
```

Comment:
Never access `BasePoints`, always go through `CalcValue()`
Also if are accessing EFFECT_1 outside of effect handler bound to that index (meaning you aren't in a function linked to `SpellEffectFn`) then it must be validated that it exists in `Validate`